pkg/manifests/external_dns_test.go

```diff
@@ -82,16 +138,208 @@ var (
+			resourceGroup:      "test-resource-group-private",
+			msiclientID:        "test-client-id",
+			serviceAccountName: "test-private-service-account",
+			namespace:          "test-namespace",
+			inputResourceName:  "test-dns-config-private",
+			identityType:       IdentityTypeWorkloadIdentity,
+			resourceTypes:      []ResourceType{ResourceTypeGateway},
+			provider:           PrivateProvider,
+			dnsZoneResourceIDs: []string{privateZoneOne, privateZoneTwo},
+			expectedLabels:     map[string]string{"app.kubernetes.io/name": "test-dns-config-private-external-dns"},
+			expectedObjects:    externalDnsResources(noOsmConf, []*ExternalDnsConfig{privateGwConfig}),
+		},
+		{
+			name:          "invalid identity type",
+			conf:          conf,
+			identityType:  3,
+			expectedError: errors.New("invalid identity type: 3"),
+		},
+		{
+			name:          "invalid resource type",
+			conf:          conf,
+			resourceTypes: []ResourceType{ResourceTypeGateway, ResourceTypeIngress, 4},
+			expectedError: errors.New("invalid resource type: 4"),
+		},
+		{
+			name:               "gateway without workload identity",
+			conf:               noOsmConf,
+			tenantId:           "test-tenant-id",
+			subscription:       "test-subscription-id",
+			resourceGroup:      "test-resource-group-private",
+			msiclientID:        "test-client-id",
+			serviceAccountName: "test-private-service-account",
+			namespace:          "test-namespace",
+			inputResourceName:  "test-resource",
+			identityType:       IdentityTypeMSI,
+			resourceTypes:      []ResourceType{ResourceTypeGateway},
+			provider:           PrivateProvider,
+			dnsZoneResourceIDs: []string{privateZoneOne, privateZoneTwo},
+			expectedError:      errors.New("gateway resource type can only be used with workload identity"),
+		},
+		{
+			name:               "workload identity without provided serviceaccount",
+			conf:               noOsmConf,
+			tenantId:           "test-tenant-id",
+			subscription:       "test-subscription-id",
+			resourceGroup:      "test-resource-group-private",
+			msiclientID:        "test-client-id",
+			serviceAccountName: "",
+			namespace:          "test-namespace",
+			inputResourceName:  "test-resource",
+			identityType:       IdentityTypeWorkloadIdentity,
+			resourceTypes:      []ResourceType{ResourceTypeIngress},
+			provider:           PrivateProvider,
+			dnsZoneResourceIDs: []string{privateZoneOne, privateZoneTwo},
+			expectedError:      errors.New("workload identity requires a service account name"),
+		},
+	}
+	for _, tc := range testCases {
+		ret, err := NewExternalDNSConfig(tc.conf, InputExternalDNSConfig{TenantId: tc.tenantId, Subscription: tc.subscription, ResourceGroup: tc.resourceGroup, ClientId: tc.msiclientID, InputServiceAccount: tc.serviceAccountName, Namespace: tc.namespace, InputResourceName: tc.inputResourceName, IdentityType: tc.identityType, ResourceTypes: tc.resourceTypes, Provider: tc.provider, DnsZoneresourceIDs: tc.dnsZoneResourceIDs})
```

Comment:
if this test struct mostly contains fields that are populating this InputExternalDNSConfig couldn't we just make a struct of this type on the test case to avoid making another layer on top of it?

Reply:
done